build.gradle

```diff
@@ -35,78 +35,64 @@ buildTypes {
     // Used for builds to run all tests, but not necessarily on all platforms
     fullTest {
         tasks "test", "forkingIntegTest", "forkingCrossVersionTest"
-        projectProperties useIncomingDistributions: true, testAllVersions: true
+        projectProperties testAllVersions: true
     }
 
     // Used for builds to test the code on certain platforms
     platformTest {
         tasks "test", "forkingIntegTest", "forkingCrossVersionTest"
-        projectProperties useIncomingDistributions: true, testAllVersions: true, testAllPlatforms: true
+        projectProperties testAllVersions: true, testAllPlatforms: true
     }
 
     // Tests not using the daemon mode
     noDaemonTest {
         tasks "noDaemonIntegTest"
-        projectProperties useIncomingDistributions: true
+        projectProperties useAllDistribution: true
```

Comment:
Is this the only place where we use the all distribution? Where do we test the real bin distribution?

Reply:
This is because we only used the `all` distribution before. We never (with the current setup) test the bin distribution outside the Build Distribution build.
But yes. It would make sense to test against `bin` as well. Although I would prefer to test the _real_ distributions built by the promote build (e.g. test them before uploading them).

Reply:
I don't think the promotion build should be the first time we test that a `bin` distribution works. The CI pipeline basically produces as its output a set of _candidate_ distributions. We should verify that each candidate is actually releasable prior to it entering the promotion build.

We definitely should _in addition_ do the same verifications as part of the promotion build on the real distributions. For this I think it's important to reuse the verifications from earlier stages rather than reimplement them.